eng/test.yaml

```diff
@@ -0,0 +1,58 @@
+steps: 
+
+- task: DotNetCoreCLI@2
+  displayName: Restore (dotnet restore)
+  inputs:
+    command: 'custom'
+    projects: |
+      src/**/*.Tests.csproj
+      !src/**/Microsoft.DotNet.Darc.Tests.csproj
+    custom: 'restore'
+  condition: succeededOrFailed()
+
+- task: DotNetCoreCLI@2
+  displayName: Test C# (dotnet test)
+  inputs:
+    command: 'custom'
+    projects: |
+      src/**/*.Tests.csproj
+      !src/**/Microsoft.DotNet.Darc.Tests.csproj
+    custom: 'test'
+    arguments: > 
+      --configuration $(_BuildConfig)
+      --collect:"Code Coverage"
+      --settings:CodeCoverage.runsettings
+      --filter "TestCategory!=PostDeployment&TestCategory!=Nightly&TestCategory!=PreDeployment"
+      --logger trx
+      --no-build
+  condition: succeededOrFailed()
+
+- powershell: |
```

Comment:
I'd rather this be a checked in file rather than inline powershell.

Reply:
I can make an issue to do that later, as this is the same functionality in Helix and I'd like to share that functionality across the projects.

Reply:
So should we change it in Helix as well? Inline powershell inside Yaml is horrible to debug.

Reply:
If the suggestion is to make this code a file inside the project, then I would assume that the same change should be made to Helix as well (and subsequent projects that make use of this code).

Reply:
I would expect this to be a script file checked in into the /eng folder, since it's a script needed for the build.

I'd rather we didn't check it in this way just because it was already done that way in Helix.

Reply:
I'll make an issue to have it changed to a file, so I can make the change to both arcade services and Helix at the same time.

Reply:
https://github.com/dotnet/core-eng/issues/8528

Reply:
What would be the negative impact of moving this to a seperate file as part of this PR @mjanecke ?

Reply:
@markwilkie Technically, I'm taking a PTO day today, so, making these changes won't be done until Monday, and I believe some of my reviewers will be OOF starting then, so I was hoping to get this PR merged before then.

Reply:
I see.  I'm fine if this slips until the new year.  We all want to get it in, but let's not (unduly) rush things.  Furthermore, you can *actually* take your day off.  :)

Reply:
I'll be here Monday :)

Reply:
Moved the script to a file.